### PR TITLE
Added Java 8 InstantParam with test.

### PR DESCRIPTION
**Pull request by lightswitch05**
Some java 8 users might want to avoid Joda-Time in favor of Instant

### CODE REVIEW

File: dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Test;
+
+import java.time.Instant;
+import java.time.ZoneOffset;
+import java.time.LocalDateTime;
+
+/**
```
Comment (joschi) [inline dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java]: Please remove the `@author` tag (or rather the complete Javadoc comment).

File: dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ * @author Daniel White
+ */
+public class InstantParamTest {
```
Comment (joschi) [inline dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java]: Please add a test case for bad input.

File: dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java (hunk 1 above)
Comment (lightswitch05) [inline dropwizard-jersey/src/test/java/io/dropwizard/jersey/params/InstantParamTest.java]: I missed that one, editor added it automatically.

**Comment by lightswitch05** [comment]
@joschi I made the requested changes, let me know if it needs more work

**Comment by joschi** [comment]
@lightswitch05 Thanks!

**Pull request merged**